ITGA7.py

```diff
@@ -0,0 +1,206 @@
+            "A": "T",
+            "C": "G",
+            "G": "C",
+            "T": "A",
+            "a": "t",
+            "c": "g",
+            "g": "c",
+            "t": "a",
+        }
+
+    def transcribe(self) -> RNASequence:
+        """
+        Returns the transcribed RNA sequence.
+        Returns:
+        - RNASequence: Transcribed RNA sequence.
+        """
+        transcribed_sequence = self.sequence.replace("T", "U").replace("t", "u")
+        return RNASequence(transcribed_sequence)
+
+
+class RNASequence(NucleicAcidSequence):
+    """
+    Represents an RNA sequence.
+    """
+
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self.ALPHABET = set("AUGCaugc")
+        self.COMPLEMENT_MAP = {
+            "A": "U",
+            "C": "G",
+            "G": "C",
+            "U": "A",
+            "a": "u",
+            "c": "g",
+            "g": "c",
+            "u": "a",
+        }
+
+
+class AminoAcidSequence(BiologicalSequence):
+    """
+    Represents an amino acid sequence.
+    """
+
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self.ALPHABET = set("GgLlYySsEeQqDdNnFfAaKkRrHhCcVvPpWwIiMmTt")
+
+    def count_aa(self) -> dict:
+        """
+        Counts the number of given or all amino acids in a protein sequence.
+        Arguments:
+        - seq (str): sequence to count amino acids
+        - aminoacids (str): which amino acids to count in sequence
+        Return:
+        - dict: a dictionary with amino acids and its count
+        """
+
+        aa_dict_count = {}
+        for aa in set(self.sequence):
+            aa_dict_count[aa] = self.sequence.count(aa)
+        return aa_dict_count
+
+
+def filter_fastq(
+    path_to_seqs: str,
+    output_file_name: str = None,
+    gc_bounds: tuple | int = (0, 100),
+    length_bounds: tuple | int = (0, 2**32),
+    quality_threshold: int = 0,
+) -> None:
+    """
+    Filters FASTQ sequences based on the GC-content, length and quality parameters.
+
+    Args:
+    - path_to_seqs (str): the path to the FASTQ file to be filtered.
+    - output_file_name (str): the name of the file where the filtered FASTQ sequences will be saved.
+    - gc_bounds (tuple, int): GC content range (in percentages) for filtering. If you pass a single number
+    to the argument, it is assumed to be an upper bound.
+    - length_bounds (tuple, int): length range for filtering. If you pass a single number to the argument,
+    it is assumed to be an upper bound.
+    - quality_threshold (int): threshold value for average read quality filtering
+
+    Returns:
+    - None: the function doesn't return a value but writes the filtered FASTQ to the output file.
+    """
+    if output_file_name is None:
+        output_file_name = os.path.basename(path_to_seqs)
+
+    records = SeqIO.parse(path_to_seqs, "fastq")
+
+    filtered_records = []
+
+    for record in records:
+        gc_content = gc_fraction(record.seq)
```

Comment:
Чуть поинтереснее было бы использовать функционал метода `BIo`, а именно класса `SeqUtils`. Там есть метод (`GC`), осуществляющий подсчет GC-состава:
```suggestion
        gc_content = Bio.SeqUtils.GC(record.seq)
```